### PR TITLE
Single integrator

### PR DESCRIPTION
**Pull request by akmaralAW**
Commit 1: Single integrator is added. Current script allows to save the output of s2sm method into .yaml file following our output.yaml format and can be run/tested with our visualization script. 
Commit 2: conversion script converts our yaml file format to s2sm environment.yaml format. Conversion script has been tested with our wall environment and two single integrators.
Commit 3: stats.yaml is saved into results/environment_name/ folder.

### CODE REVIEW

File: conversion.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        per_robot = {}
+        per_robot["k"] = [0.5, 0.5] # 2D for single integrator
+        per_robot["size"] = 1.5
+        per_robot["type"] = robots[i]["type"]
+        per_robot["velocity"] = 1.0
+        new_format["agents"].append(per_robot)
+        new_format["goals"].append(get_polytope(robots[i]["goal"]))
+        new_format["starts"].append(robots[i]["start"])
+    for j in range(len(obstacles)):
+        new_format["obstacles"].append(get_polytope(obstacles[j]["center"])) # size ?
```
Comment (whoenig) [inline conversion.py:42]: don't obstacles have a size in our yaml format?

File: models/single_integrator.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        self.size = size
+        self.velocity = velocity
+        self.k = k
+
+    def model(self, q, t, u):
+        x, y = q
+        vx, vy = u
+        dxdt = vx 
+        dydt = vy 
+        return [dxdt, dydt]
```
Comment (whoenig) [inline models/single_integrator.py]: can be simplified to `return u`?

File: models/single_integrator.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        x, y = q
+        xref, yref = qref
+        vxref, vyref = uref
+        k1, k2  = self.k
+        # needs to be checked
+        xe = xref - x
+        ye = yref - y
+        vx = vxref + k1 * xe
+        vy = vyref * k2 * ye
+        return [vx, vy]
```
Comment (whoenig) [inline models/single_integrator.py]: Use vector computation. uref + self.k.dot(qref-q). Also note that you still have the typo in l. 28

File: models/single_integrator.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    def bloating(self, n):
+        return 0
+        k1, k2 = self.k
+        if n != 0:
+            return sqrt(4 * n / k2)
+        else:
+            return sqrt(4 / k2)
```
Comment (whoenig) [inline models/single_integrator.py:25]: what does this function do?

File: problems/zigzag/problem.yaml [hunk 5]
```diff
@@ -3,18 +3,18 @@ agents:
     size: 1.5
```
Comment (whoenig) [inline problems/zigzag/problem.yaml]: avoid changing an existing problem.yaml. Create a new problem instead.

**Review by whoenig — CHANGES_REQUESTED** [review]

File: conversion.py (hunk 1 above)
Comment (akmaralAW) [reply conversion.py:42]: yes, we have. But s2sm takes the center of the obstacle and creates the Polytope.

File: models/single_integrator.py (hunk 4 above)
Comment (akmaralAW) [reply models/single_integrator.py:25]: function is called while finding collision, but it returns 0 all the time, right ?

File: conversion.py (hunk 1 above)
Comment (whoenig) [reply conversion.py:42]: Yes, but when we create the polytope, we should use the size (as epsilon on how we shift the 4 planes).

File: models/single_integrator.py (hunk 4 above)
Comment (whoenig) [reply models/single_integrator.py:25]: I didn't see that part, I guess we should delete the dead code then. However, what is that function supposed to do and what is n?

File: problems/zigzag/problem.yaml (hunk 5 above)
Comment (whoenig) [reply problems/zigzag/problem.yaml]: Like before, try to not change this existing file

**Review by whoenig — COMMENTED** [review]
Much better - thanks. A few more open discussion points.

**Pull request merged**